### PR TITLE
Docs: ILM document behaviour for changing lifecycle setting

### LINKED ISSUE #70151: Changing the ILM policy for a managed index is non-deterministic
Currently, assigning a managed index to a different policy yields a non-deterministic behaviour due to the caching of the current phase an index is going through.

eg. An index managed by an ILM policy (policy A) is added to a new ILM policy (policy B) while still in the hot phase of policy A by using the [update index settings API](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html). Due to the cached phase definition in the index metadata, even if that phase is from a previous ILM policy (policy A), it will still need to be completed. Therefore the index will remain in the hot phase pending to roll over, having to wait for the rollover of the first policy to happen before it can go into a new phase (the new phase in this case will be the first phase of policy B)

This would not be the case if the [ILM remove policy API](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-remove-policy.html) would be used to remove the first policy, before assigning a new one, however, this api might be something our users don't know of (as opposed to the update index settings API)

I'm proposing to at least document the non-deterministic behaviour and/or react to the `index.lifecycle.name` configuration change and mimic the [ILM remove policy API](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-remove-policy.html) behaviour.

### PR DESCRIPTION
**Pull request by andreidan**
Closes #70151

### CODE REVIEW

File: docs/reference/settings/ilm-settings.asciidoc [hunk 1]
```diff
@@ -37,8 +37,11 @@ You can explicitly set it to <<skipping-rollover, skip rollover>>.
 Defaults to `false`.
 
 `index.lifecycle.name`::
-(<<indices-update-settings, Dynamic>>, string) 
-The name of the policy to use to manage the index.
+(<<indices-update-settings, Dynamic>>, string)
+The name of the policy to use to manage the index. As the current executing phase
+definition is cached in the index metadata, changing the setting to a different policy
+will not affect the currently executing phase, but the next phase will be read from
+the newly configured policy.
```
Comment (jrodewig) [inline docs/reference/settings/ilm-settings.asciidoc:44]: This seems to say the same thing as https://www.elastic.co/guide/en/elasticsearch/reference/master/update-lifecycle-policy.html#ilm-apply-new-policy. I wonder if we should just add an xref to there?

**Review by jrodewig — COMMENTED** [review]
Thanks @andreidan. The changes look good but may be redundant. WDYT?

File: docs/reference/settings/ilm-settings.asciidoc (hunk 1 above)
Comment (andreidan) [reply docs/reference/settings/ilm-settings.asciidoc:44]: @jrodewig great point! I missed that we document this already. An xref sounds great

Comment (jrodewig) [reply docs/reference/settings/ilm-settings.asciidoc]: Awesome. I pushed 28f7fce15e4a59ab8c9a69939b731929503a89eb to add an xref.

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @andreidan.

**Pull request merged**